### ISSUE TITLE
Install error

### ISSUE THREAD

**Issue by mdaniel29** [osxmidi/LinVst3#11]
Hi.
Ive tried to compile LinVst3 from sources but I get an error.
I followed the instructions on Readme.md, installes every dependencies, downloaded the VST3 SDK, launched the "make" into the LinVst3 directory (which is within the VST3_SDK folder) but I get :
**"/home/user/dev/VST_SDK/VST3_SDK/base/source/fdynlib.cpp:43:10:** fatal error: Windows.h: Aucun fichier ou dossier de ce type"
I tried to rename Windows.h to windows.h but the problem is the same.
**sudo find / -name windows.h -print** returns : 
/usr/i686-w64-mingw32/include/windows.h
/usr/x86_64-w64-mingw32/include/windows.h
/usr/share/mingw-w64/include/windows.h

Any help would be appreciated - Thanks.

**Comment by osxmidi** [comment]
> Hi.
> Ive tried to compile LinVst3 from sources but I get an error.
> I followed the instructions on Readme.md, installes every dependencies, downloaded the VST3 SDK, launched the "make" into the LinVst3 directory (which is within the VST3_SDK folder) but I get :
> **"/home/user/dev/VST_SDK/VST3_SDK/base/source/fdynlib.cpp:43:10:** fatal error: Windows.h: Aucun fichier ou dossier de ce type"
> I tried to rename Windows.h to windows.h but the problem is the same.
> **sudo find / -name windows.h -print** returns :
> /usr/i686-w64-mingw32/include/windows.h
> /usr/x86_64-w64-mingw32/include/windows.h
> /usr/share/mingw-w64/include/windows.h
> 
> Any help would be appreciated - Thanks.

Hi,

I think that it's because windows.h can't be found on your system.

The windows.h file should be Wines development windows.h file.

You seem to only have windows.h from mingw which is not the windows.h to use.

Install details are at https://github.com/osxmidi/LinVst/tree/master/Make-Guide

For Debian for example these would need to be installed, 

sudo apt-get install libx11-dev

sudo apt-get install wine-stable-dev or sudo apt-get install wine-staging-dev

**Comment by mdaniel29** [comment]
=== 21/02/21
I uninstalled all my wine-staging packages, so I could install libwine-dev, libwine-development, libwine-development-dev
Next I had the same error messages when I compiled with make (Windows.h, so such file or directory...)
So I renamed "Windows.h" to "windows.h" in all these files :
fdynlib.cpp, fstring.cpp, timer.cpp, fcondition.cpp, flock.cpp, systemclipboard_win32.cpp, threadchecker_win32.cpp
No complains about Missing Windows.h anymore.
I also had to modify the line about memcpy_s ("was not declared in this scope") > I changed to memcpy, with the right arguments

In the end, still an error :
lin-vst-server.cpp:(.text+0xb2cb): undefined reference to `Steinberg::IPlugFrame::iid'

And now I don't know what to do.

What is wrong with my install : a bad / non supported wine version ? (wine package delivered with Debian Buster distro)

Thanks.




----- Mail original -----
De: "osxmidi" <notifications@github.com>
À: "osxmidi/LinVst3" <LinVst3@noreply.github.com>
Cc: "homer00" <ami8@free.fr>, "Author" <author@noreply.github.com>
Envoyé: Vendredi 19 Février 2021 06:49:12
Objet: Re: [osxmidi/LinVst3] Install error (#11)







Hi. 
Ive tried to compile LinVst3 from sources but I get an error. 
I followed the instructions on Readme.md, installes every dependencies, downloaded the VST3 SDK, launched the "make" into the LinVst3 directory (which is within the VST3_SDK folder) but I get : 
"/home/user/dev/VST_SDK/VST3_SDK/base/source/fdynlib.cpp:43:10: fatal error: Windows.h: Aucun fichier ou dossier de ce type" 
I tried to rename Windows.h to windows.h but the problem is the same. 
sudo find / -name windows.h -print returns : 
/usr/i686-w64-mingw32/include/windows.h 
/usr/x86_64-w64-mingw32/include/windows.h 
/usr/share/mingw-w64/include/windows.h 

Any help would be appreciated - Thanks. 

Hi, 

I think that it's because windows.h can't be found on your system. 

The windows.h file should be Wines development windows.h file. 

You seem to only have windows.h from mingw which is not the windows.h to use.

**Comment by osxmidi** [comment]
> === 21/02/21 I uninstalled all my wine-staging packages, so I could install libwine-dev, libwine-development, libwine-development-dev Next I had the same error messages when I compiled with make (Windows.h, so such file or directory...) So I renamed "Windows.h" to "windows.h" in all these files : fdynlib.cpp, fstring.cpp, timer.cpp, fcondition.cpp, flock.cpp, systemclipboard_win32.cpp, threadchecker_win32.cpp No complains about Missing Windows.h anymore. I also had to modify the line about memcpy_s ("was not declared in this scope") > I changed to memcpy, with the right arguments In the end, still an error : lin-vst-server.cpp:(.text+0xb2cb): undefined reference to `Steinberg::IPlugFrame::iid' And now I don't know what to do. What is wrong with my install : a bad / non supported wine version ? (wine package delivered with Debian Buster distro) Thanks.
> […](#)
> ----- Mail original ----- De: "osxmidi" <notifications@github.com> À: "osxmidi/LinVst3" <LinVst3@noreply.github.com> Cc: "homer00" <ami8@free.fr>, "Author" <author@noreply.github.com> Envoyé: Vendredi 19 Février 2021 06:49:12 Objet: Re: [osxmidi/LinVst3] Install error (#11) Hi. Ive tried to compile LinVst3 from sources but I get an error. I followed the instructions on Readme.md, installes every dependencies, downloaded the VST3 SDK, launched the "make" into the LinVst3 directory (which is within the VST3_SDK folder) but I get : "/home/user/dev/VST_SDK/VST3_SDK/base/source/fdynlib.cpp:43:10: fatal error: Windows.h: Aucun fichier ou dossier de ce type" I tried to rename Windows.h to windows.h but the problem is the same. sudo find / -name windows.h -print returns : /usr/i686-w64-mingw32/include/windows.h /usr/x86_64-w64-mingw32/include/windows.h /usr/share/mingw-w64/include/windows.h Any help would be appreciated - Thanks. Hi, I think that it's because windows.h can't be found on your system. The windows.h file should be Wines development windows.h file. You seem to only have windows.h from mingw which is not the windows.h to use. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub , or unsubscribe .

The only windows.h that I can find on my system (MX Linux Debian Buster) is 

/opt/wine-staging/include/wine/windows/windows.h

I don't have any Windows.h files and Windows.h (with an uppercase W) should not be the one that the Makefile is selecting, so something is wrong there.

There is no need to edit the Makefile, the Makefile should be automatically selecting windows.h (with a lower case w) if your system Wine installs are ok.

What I do is (for Debian, but there are similar instructions for Manjaro etc),

Install the WineHQ Wine-Stable or Wine-Staging from the WineHQ site.

then,

sudo apt-get install libx11-dev

sudo apt-get install wine-stable-dev or sudo apt-get install wine-staging-dev (these contain windows.h)

and then vst3 also needs,

sudo apt-get install cmake

sudo apt-get install libfreetype6-dev libxcb-util0-dev libxcb-cursor-dev libxcb-keysyms1-dev libxcb-xkb-dev libxkbcommon-dev libxkbcommon-x11-dev libgtkmm-3.0-dev libsqlite3-dev

and also the vst3 sdk from Steinberg.

----------------------

For optional 32 bit vsts (doesn't apply for LinVst3) these also need to be installed

sudo dpkg --add-architecture i386

sudo apt-get install libc6-dev-i386

sudo apt-get install gcc-multilib g++-multilib

**Comment by osxmidi** [comment]
I found out what your errors are.

I tried the new vst3 sdk from Steinberg on Manjaro and I got the same Windows.h error.

It's because Steinberg updated the vst3 sdk and I havn't got around to updating LinVst3 for it.

The vst3 sdk to use is the previous version Currently builds ok with the vst-sdk_3.7.0_build-116_2020-07-31 sdk https://download.steinberg.net/sdk_downloads/vst-sdk_3.7.0_build-116_2020-07-31.zip

Also the windows.h file on Manjaro is located at /usr/include/wine/windows/windows.h and I'll update the Makefile for Manjaro users.
